mrp_lot_number_propagation/tests/test_mrp_production.py

```diff
@@ -143,3 +175,87 @@ def test_confirm_with_variant_no(self):
                     new_order.action_confirm()
             else:
                 new_order.action_confirm()
+
+    def test_mass_produce_wizard(self):
+        order = self._create_order(self.bom_product_product, self.bom, qty=5)
+        self._update_stock_component_qty(order)
+        order.action_confirm()
+        tracked_moves = order.move_raw_ids.filtered(
+            lambda mv: mv.product_id.tracking != "none"
+        )
+        self.assertTrue(len(tracked_moves) > 1)
+        res = order.button_mark_done()
+        self.assertEqual(type(res), dict)
+        self.assertEqual(res["res_model"], "mrp.batch.produce")
+
+        propagate_move = order._get_propagating_component_move()
+        propagate_move_lines = propagate_move.move_line_ids
+        component_lot_name = (tracked_moves - propagate_move).move_line_ids.lot_id.name
+
+        wizard_form = Form(self.env["mrp.batch.produce"].with_context(**res["context"]))
+
+        # Executing wizard without entering same name for finished product and
+        #  propagated component must raise an error
+        def random_name():
+            return "".join(random.choice(string.ascii_lowercase) for i in range(10))
+
+        wizard_form.production_text = "\n".join(
+            [
+                f"{random_name()},{comp_serial_name},{component_lot_name}"
+                for comp_serial_name in propagate_move_lines.lot_id.mapped("name")
+            ]
+        )
+        wizard = wizard_form.save()
+        with self.assertRaisesRegex(
+            UserError, "set to propagate lot number from component"
+        ):
+            wizard.action_done()
+        wizard_form = Form(wizard)
+        wizard_form.production_text = "\n".join(
+            [
+                f"{comp_serial_name},{comp_serial_name},{component_lot_name}"
```

Comment:
why the serial should be duplicated?

Reply:
The module is meant to propagate lot number from a component to the finished product. In that case we need to define it twice, first for the finished product, then for the component that is propagating.